uwblueprint/focus-on-nature
# feat: remove camper from waitlist

## Notion ticket link

[Camp Overview: remove camper from waitlist](https://www.notion.so/uwblueprintexecs/Camp-Overview-remove-camper-from-waitlist-5ad1ca0540d84028afa311a1eb6d9b9f)


## Implementation description

NOTE** this PR branches off this one: https://github.com/uwblueprint/focus-on-nature/pull/106
* added `deleteCamperFromWaitlist` function to `CamperAPIClient`
* added `GeneralDeleteCamperModal` component to common 
* implemented deletion in waitlisted camper table

![Screen Shot 2022-09-17 at 4 33 05 PM](https://user-images.githubusercontent.com/46568041/190875521-356d7cc8-f4ac-4e34-81ef-85531d9dcb30.png)


## Steps to test
1. go to a camps overview page e.g. http://localhost:3000/admin/camp/630a63450eb59852f9d01b1e
2. go to the waitlist table tab at the bottom of the page 
3. click on the three-dot button and select "Remove"
4. click "Remove" on the modal 
5. check that the camper was removed from database here https://cloud.mongodb.com/v2/61ebbfc0866f7f3a496b1389#metrics/replicaSet/61ebc0e00a9a4c02dfab887f/explorer/focusOnNature/waitlistedcampers/find


## What should reviewers focus on?
* does the styling match the design enough 
* feature functionality


## Checklist
